lib/teleterm/daemon/daemon.go

```diff
@@ -89,6 +88,7 @@ func New(cfg Config) (*Service, error) {
 		gateways:               make(map[string]gateway.Gateway),
 		usageReporter:          connectUsageReporter,
 		headlessWatcherClosers: make(map[string]context.CancelFunc),
+		headlessAuthSemaphore:  newWaitSemaphore(maxConcurrentImportantModals, imporantModalWaitDuraiton),
 	}
```

Comment:
What's this chance for? It seems unrelated

Reply:
This was previously done in `UpdateAndDialTshdEventsServerAddress` which connects the client. It should have just been initialized here in the first place.